### PR TITLE
🔧 MAINTAIN: Pin tests requirements to functional versions.

### LINKED ISSUE #227: Tests fail to execute with AssertionError
Tests are currently failing to execute with an `AssertionError`:

```bash
AssertionError
ERROR: InvocationError for command /home/sadorf/local/csadorf/plumpy/.tox/py37/bin/pytest (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   py37: commands failed
```

<details>
<summary>Full traceback</summary>

```bash
py37 develop-inst-noop: /home/sadorf/local/csadorf/plumpy
py37 installed: aio-pika==6.8.2,aiormq==3.3.1,anyio==3.5.0,argon2-cffi==21.3.0,argon2-cffi-bindings==21.2.0,async-generator==1.10,attrs==21.4.0,backcall==0.2.0,bleach==4.1.0,certifi==2021.10.8,cffi==1.15.0,charset-normalizer==2.0.12,colorama==0.4.4,coverage==6.3.2,debugpy==1.5.1,decorator==5.1.1,defusedxml==0.7.1,deprecation==2.1.0,entrypoints==0.4,gitdb==4.0.9,GitPython==3.1.27,idna==3.3,importlib-metadata==4.11.1,importlib-resources==5.4.0,ipykernel==6.9.1,ipython==7.31.1,ipython-genutils==0.2.0,jedi==0.18.1,Jinja2==3.0.3,jsonschema==4.4.0,jupyter-client==6.1.12,jupyter-core==4.9.2,jupyter-server==1.13.5,jupyter-server-mathjax==0.2.5,jupyterlab-pygments==0.1.2,kiwipy==0.7.5,MarkupSafe==2.1.0,matplotlib-inline==0.1.3,mistune==0.8.4,more-itertools==8.12.0,multidict==6.0.2,nbclient==0.5.11,nbconvert==6.4.2,nbdime==3.1.1,nbformat==5.1.3,nest-asyncio==1.4.3,packaging==21.3,pamqp==2.3.0,pandocfilters==1.5.0,parso==0.8.3,pexpect==4.8.0,pickleshare==0.7.5,pluggy==0.13.1,-e git+https://github.com/csadorf/plumpy.git@8bf5ffa09fe0da7fc777991970a7658851ddb36a#egg=plumpy,prometheus-client==0.13.1,prompt-toolkit==3.0.28,ptyprocess==0.7.0,py==1.11.0,pycparser==2.21,Pygments==2.11.2,pyparsing==3.0.7,pyrsistent==0.18.1,pytest==5.4.3,pytest-asyncio==0.17.0,pytest-cov==3.0.0,pytest_notebook==0.7.0,python-dateutil==2.8.2,pytray==0.3.2,PyYAML==5.4.1,pyzmq==22.3.0,requests==2.27.1,Send2Trash==1.8.0,shortuuid==1.0.8,six==1.16.0,smmap==5.0.0,sniffio==1.2.0,terminado==0.13.1,testpath==0.6.0,tomli==2.0.1,tornado==6.1,traitlets==5.1.1,typing_extensions==4.1.1,urllib3==1.26.8,wcwidth==0.2.5,webencodings==0.5.1,websocket-client==1.2.3,yarl==1.7.2,zipp==3.7.0
py37 run-test-pre: PYTHONHASHSEED='3792903192'
py37 run-test: commands[0] | pytest
Traceback (most recent call last):
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/bin/pytest", line 8, in <module>
    sys.exit(main())
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/_pytest/config/__init__.py", line 105, in main
    config = _prepareconfig(args, plugins)
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/_pytest/config/__init__.py", line 258, in _prepareconfig
    pluginmanager=pluginmanager, args=args
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/pluggy/callers.py", line 203, in _multicall
    gen.send(outcome)
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/_pytest/helpconfig.py", line 90, in pytest_cmdline_parse
    config = outcome.get_result()
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/_pytest/config/__init__.py", line 836, in pytest_cmdline_parse
    self.parse(args)
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/_pytest/config/__init__.py", line 1044, in parse
    self._preparse(args, addopts=addopts)
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/_pytest/config/__init__.py", line 992, in _preparse
    self.pluginmanager.load_setuptools_entrypoints("pytest11")
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/pluggy/manager.py", line 300, in load_setuptools_entrypoints
    self.register(plugin, name=ep.name)
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/_pytest/config/__init__.py", line 379, in register
    ret = super().register(plugin, name)
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/pluggy/manager.py", line 127, in register
    hook._maybe_apply_history(hookimpl)
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/pluggy/hooks.py", line 333, in _maybe_apply_history
    res = self._hookexec(self, [method], kwargs)
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/pytest_asyncio/plugin.py", line 57, in pytest_addoption
    default="legacy",
  File "/home/sadorf/local/csadorf/plumpy/.tox/py37/lib/python3.7/site-packages/_pytest/config/argparsing.py", line 177, in addini
    assert type in (None, "pathlist", "args", "linelist", "bool")
AssertionError
ERROR: InvocationError for command /home/sadorf/local/csadorf/plumpy/.tox/py37/bin/pytest (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   py37: commands failed
```

### PR DESCRIPTION
**Pull request by csadorf**
Fixes #227.

### CODE REVIEW

**Comment by csadorf** [comment]
I would recommend to rebase and merge this once approved.

**Comment by csadorf** [comment]
@sphuber Once you agree, I will remove `tests (3.6)` from the expected tests.

**Review by sphuber — APPROVED** [review]
Thanks @csadorf

**Pull request merged**
